Mathlib/RingTheory/IntegrallyClosed.lean

```diff
@@ -117,29 +168,38 @@ theorem integralClosure_eq_bot_iff : integralClosure R K = ⊥ ↔ IsIntegrallyC
 variable (R)
 
 @[simp]
-theorem integralClosure_eq_bot : integralClosure R K = ⊥ :=
-  (integralClosure_eq_bot_iff K).mpr ‹_›
+theorem integralClosure_eq_bot : integralClosure R A = ⊥ :=
+  (integralClosure_eq_bot_iff A).mpr ‹_›
 #align is_integrally_closed.integral_closure_eq_bot IsIntegrallyClosed.integralClosure_eq_bot
 
-end IsIntegrallyClosed
 
-namespace integralClosure
+end Algebra
 
-open IsIntegrallyClosed
+section Injective
 
-variable {R : Type*} [CommRing R]
+-- TODO: this section can be generalized from `IsFractionRing` to any `Algebra` that is injective,
+-- but there is no way to hook that assumption up to the typeclass system.
```

Comment:
iirc, it's `NoZeroSmulDivisors`

Reply:
The identity homomorphism from R to itself is injective but wouldn't satisfy `NoZeroSmulDivisors` if R has zero divisors.

Reply:
ah, indeed, I just remember using it in flt-reg for that purpose but that's just because we were dealing with domains at all times